packages/ui/src/SearchDialog.tsx

```diff
@@ -28,6 +32,61 @@ export function SearchDialog({ tracks }: { tracks: (Track & { problems: Problem[
     };
   }, []);
 
+  useEffect(() => {
+    const handleKeyPress = (event: KeyboardEvent) => {
+      if (event.code === "ArrowDown") {
+        event.preventDefault();
+
+        const list = searchFilterListRef.current;
+        if (list && selectedTrack + 1 < searchTracks.length) {
+          const currentItem = list.children[selectedTrack + 1];
+          if ((list[selectedTrack + 1] && !currentItem.offsetHeight) || currentItem.offsetTop > list.offsetHeight) {
+            currentItem.scrollIntoView({ behavior: "smooth", block: "end" });
+          }
+        }
+
+        setSelectedTrack((prev) => {
+          if (prev < searchTracks.length - 1) {
+            return prev + 1;
+          }
+          return prev;
+        });
+      }
+      if (event.code === "ArrowUp") {
+        event.preventDefault();
+
+        const list = searchFilterListRef.current;
+        if (list && selectedTrack - 1 > -1) {
+          const currentItem = list.children[selectedTrack - 1];
+          if ((list[selectedTrack - 1] && !currentItem.offsetHeight) || currentItem.offsetTop < list.offsetHeight) {
+            currentItem.scrollIntoView({ behavior: "smooth", block: "start" });
+          }
+        }
+
+        setSelectedTrack((prev) => {
+          if (prev > 0) {
+            return prev - 1;
+          }
+          return prev;
+        });
+      }
+      if (event.code === "Enter") {
+        event.preventDefault();
+        router.push(`/tracks/${searchTracks[selectedTrack].id}/${searchTracks[selectedTrack].problems[0]?.id}`);
+      }
+    };
+
+    window.addEventListener("keydown", handleKeyPress);
+
+    return () => {
+      window.removeEventListener("keydown", handleKeyPress);
+    };
+  }, [searchTracks, selectedTrack]);
```

Comment:
This'll reload everytime `selectedTrack` changes. Is there a better way of doing it without removing and re-attaching the handler?
Not a priority

Reply:
figure this